Bosses.py

```diff
@@ -26,18 +27,19 @@ def ArmosKnightsDefeatRule(state, player):
         state.has('Red Boomerang', player))
 
 def LanmolasDefeatRule(state, player):
-    # TODO: Allow the canes here?
     return (
         state.has_blunt_weapon(player) or
         state.has('Fire Rod', player) or
         state.has('Ice Rod', player) or
+        state.has('Cane of Somaria', player) or
+        state.has('Cane of Byrna', player) or
         state.can_shoot_arrows(player))
 
 def MoldormDefeatRule(state, player):
     return state.has_blunt_weapon(player)
 
 def HelmasaurKingDefeatRule(state, player):
-    return state.has_blunt_weapon(player) or state.can_shoot_arrows(player)
+    return state.has_sword(player) or state.can_shoot_arrows(player)
```

Comment:
Why remove the Hammer?

Reply:
Because trying to hit him in phase 2 with just the hammer sucks, and isn't in logic on the website.  You need hammer or bombs for phase 1, but you always have bombs.

Reply:
Yes, we'll go with website logic on this one for consistency.